.github/workflows/build.yml

```diff
@@ -123,19 +103,24 @@ jobs:
       - name: Run Archetype Tests
         run: |
           ./mvnw -B clean install -Parchetype-test -pl :foundation-archetype
-      #NB: The following two explicit cache saves are necessary to ensure caches are saved on build failure,
-      # until https://github.com/actions/cache/issues/1315 is resolved
```

Comment:
I: This issue is unlikely to be resolved any time soon as there are underlying restrictions that prevent it.  Instead, we follow [the guidance](https://github.com/actions/cache/issues/1315#issuecomment-2163822166) of the action maintainers.